web/packages/design/src/DataTable/useTable.ts

```diff
@@ -101,15 +105,28 @@ export default function useTable<T>({
           showFirst
         );
     if (pagination && !serversideProps) {
+      const pages = paginateData(sortedAndFiltered, pagination.pageSize);
+      // Preserve the current page, instead of resetting it every data update.
+      // The currentPage index can be out of range if data were deleted
+      // from the original data. If that were the case, reset the currentPage
+      // to the last page.
+      let currentPage = state.pagination.currentPage;
+      if (resetCurrentPage) {
+        // Resetting the current page is desirable when user is newly sorting
+        // or entered a new search.
+        currentPage = 0;
+      } else if (currentPage && pages.length > 0 && !pages[currentPage]) {
+        currentPage = pages.length - 1;
+      }
```

Comment:
what happens if multiple pages of content is deleted at the same time? i guess this would just go "to the last page" regardless if something liek 4 were deleted so thats fine